core/src/kvs/surrealkv/mod.rs

```diff
@@ -0,0 +1,398 @@
+			return Err(Error::TxReadonly);
+		}
+		// Mark this transaction as done
+		self.done = true;
+
+		// Commit this transaction
+		self.inner.commit().await?;
+		// Continue
+		Ok(())
+	}
+	/// Check if a key exists
+	pub(crate) async fn exi<K>(&mut self, key: K) -> Result<bool, Error>
+	where
+		K: Into<Key>,
+	{
+		// Check to see if transaction is closed
+		if self.done {
+			return Err(Error::TxFinished);
+		}
+		// Check the key
+		self.inner
+			.get(&key.into().as_slice())
+			.map(|opt| opt.is_some())
+			.map_err(|e| Error::Tx(format!("Unable to get kv from SurreaKV: {}", e)))
+	}
+	/// Fetch a key from the database
+	pub(crate) async fn get<K>(&mut self, key: K) -> Result<Option<Val>, Error>
+	where
+		K: Into<Key>,
+	{
+		// Check to see if transaction is closed
+		if self.done {
+			return Err(Error::TxFinished);
+		}
+		// Get the key
+		let res = self.inner.get(&key.into().as_slice())?;
+
+		// Return result
+		Ok(res)
+	}
+	/// Obtain a new change timestamp for a key
+	/// which is replaced with the current timestamp when the transaction is committed.
+	/// NOTE: This should be called when composing the change feed entries for this transaction,
+	/// which should be done immediately before the transaction commit.
+	/// That is to keep other transactions commit delay(pessimistic) or conflict(optimistic) as less as possible.
+	#[allow(unused)]
+	pub(crate) async fn get_timestamp<K>(&mut self, key: K) -> Result<Versionstamp, Error>
+	where
+		K: Into<Key>,
+	{
+		// Check to see if transaction is closed
+		if self.done {
+			return Err(Error::TxFinished);
+		}
+		// Write the timestamp to the "last-write-timestamp" key
+		// to ensure that no other transactions can commit with older timestamps.
+		let key_vec = key.into();
+		let k = key_vec.as_slice();
+		let prev = self.inner.get(k.clone())?;
+
+		let ver = match prev {
+			Some(prev) => {
+				let slice = prev.as_slice();
+				let res: Result<[u8; 10], Error> = match slice.try_into() {
+					Ok(ba) => Ok(ba),
+					Err(e) => Err(Error::Ds(e.to_string())),
+				};
+				let array = res?;
+				let prev: u64 = try_to_u64_be(array)?;
+				prev + 1
+			}
+			None => 1,
+		};
+
+		let verbytes = u64_to_versionstamp(ver);
+
+		self.inner.set(k, verbytes.as_slice())?;
+		// Return the uint64 representation of the timestamp as the result
+		Ok(verbytes)
+	}
+	/// Obtain a new key that is suffixed with the change timestamp
+	pub(crate) async fn get_versionstamped_key<K>(
+		&mut self,
+		ts_key: K,
+		prefix: K,
+		suffix: K,
+	) -> Result<Vec<u8>, Error>
+	where
+		K: Into<Key>,
+	{
+		// Check to see if transaction is closed
+		if self.done {
+			return Err(Error::TxFinished);
+		}
+		// Check to see if transaction is writable
+		if !self.write {
+			return Err(Error::TxReadonly);
+		}
+		let ts = self.get_timestamp(ts_key).await?;
+		let mut k: Vec<u8> = prefix.into();
+		k.append(&mut ts.to_vec());
+		k.append(&mut suffix.into());
+		Ok(k)
+	}
+	/// Insert or update a key in the database
```

Comment:
Ive noticed that all these functions don't have new lines preceding the rustdoc comment. We should have spacing. I think this can be achieved with `cargo fmt` which will update all the files - worth running that. I have it attached to "on save" in my editor.

```suggestion

	/// Insert or update a key in the database
```

Reply:
Thank you, I'll just fix this